core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -190,6 +190,12 @@ class DAGScheduler(
   /**
    * Number of consecutive stage attempts allowed before a stage is aborted.
    */
+  private[scheduler] val unRegisterOutputOnHostOnFetchFailure =
+    sc.getConf.getBoolean("spark.fetch.failure.unRegister.output.on.host", true)
```

Comment:
How about rename it to `spark.files.fetchFailure.unRegisterOutputOnHost` and make it default to false?